server/impl/src/main/java/com/walmartlabs/concord/server/user/UserManager.java

```diff
@@ -230,7 +219,18 @@ public void disable(UUID userId) {
                 .log();
     }
 
-    private UserInfoProvider assertProvider(UserType type) {
+    private UserInfoProvider assertProvider(String username, UserType type) {
+        /* Override SSO type if current user loggedIn via SSO */
+        if (!type.equals(UserType.SSO)){
```

Comment:
Why do we need this? This is some strange logic to ask for a `UserInfoProvider` of the specified type and then return something else.

Reply:
This override is needed because to address getInfo from [here](https://github.com/walmartlabs/concord/blob/5dd4403ca4f9e65251d7c67bc5e0d09d2e94048c/server/impl/src/main/java/com/walmartlabs/concord/server/process/pipelines/processors/UserInfoProcessor.java#L52) and [here](https://github.com/walmartlabs/concord/blob/5dd4403ca4f9e65251d7c67bc5e0d09d2e94048c/server/impl/src/main/java/com/walmartlabs/concord/server/policy/PolicyManager.java#L133) otherwise will get following error although process is trigger via UI after SSO login

![image](https://user-images.githubusercontent.com/70886332/109378782-3bc7b400-78fb-11eb-915b-cd930bb00d6b.png)

Now, I have refactored this logic to separate method #assertUserType